src/utils.ts

```diff
@@ -1,15 +1,52 @@
-import type { TokenScore } from "./types/index.js";
+import type { Document, SearchableValue, TokenScore } from "./types.js";
 
 const baseId = Date.now().toString().slice(5);
 let lastId = 0;
 
+// Checks if `hasOwn` method is defined avoiding errors with older Node.js versions
+const hasOwn = Object.hasOwn ?? Object.prototype.hasOwnProperty.call;
+
 const k = 1024;
 const nano = BigInt(1e3);
 const milli = BigInt(1e6);
 const second = BigInt(1e9);
 
 export const isServer = typeof window === "undefined";
 
+export function sprintf(template: string, ...args: (string | number)[]): string {
```

Comment:
You crazy guy

Reply:
Lol. In my defense I already had it in another toy project so I just copied it. XD